src/com/amazon/corretto/crypto/provider/HkdfRfc5869Spec.java

```diff
@@ -0,0 +1,152 @@
+// Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+package com.amazon.corretto.crypto.provider;
+
+import java.security.spec.KeySpec;
+
+public class HkdfRfc5869Spec implements KeySpec {
+  final int mode;
+  final byte[] secretOrPrk;
+  final byte[] salt;
+  final byte[] info;
+  final int desiredLength;
+  final String algorithmName;
+  public static final String DEFAULT_ALGORITHM_NAME = "HkdfRfc5869";
+  public static final int HKDF_MODE = 1;
+  public static final int HKDF_EXTRACT_MODE = 2;
+  public static final int HKDF_EXPAND_MODE = 3;
+
+  public HkdfRfc5869Spec(
+      final int mode,
+      final byte[] secret,
+      final byte[] salt,
+      final byte[] info,
+      final byte[] prk,
+      final int desiredLength,
+      final String algorithmName) {
+
+    switch (mode) {
+      case HKDF_MODE:
+        this.secretOrPrk = Utils.requireNonNull(secret, "secret cannot be null for HKDF");
+        this.salt = Utils.requireNonNull(salt, "salt cannot be null for HKDF");
+        this.info = Utils.requireNonNull(info, "info cannot be null for HKDF");
+        this.desiredLength = validateDesiredLength(desiredLength);
+        break;
+      case HKDF_EXTRACT_MODE:
+        this.secretOrPrk = Utils.requireNonNull(secret, "secret cannot be null for HKDF_EXTRACT");
+        this.salt = Utils.requireNonNull(salt, "salt cannot be null for HKDF_EXTRACT");
+        this.info = null;
+        this.desiredLength = 0;
+        break;
+      case HKDF_EXPAND_MODE:
+        this.secretOrPrk = Utils.requireNonNull(prk, "prk cannot be null for HKDF_EXPAND");
```

Comment:
Should we check that the PRK length is at least the size of the digest? From RFC5869

> PRK      a pseudorandom key of at least HashLen octets

Reply:
LC is fine with empty input.